components/automate-ui/src/app/entities/projects/project.effects.ts

```diff
@@ -151,5 +162,41 @@ export class ProjectEffects {
           message: `Could not update project: ${msg || payload.error}`
         });
       }));
+
+  @Effect()
+  applyRulesStart$ = this.actions$.pipe(
+    ofType<ApplyRulesStart>(ProjectActionTypes.APPLY_RULES_START),
+    mergeMap(() =>
+      this.requests.applyRulesStart().pipe(
+        switchMap(() => [
```

Comment:
Help me understand what `switchMap` is doing here. (I am familiar with the operator; talking about this specific implementation. 😁 ) I am guessing that since its anonymous function is returning a list of two Observables it emits on the first (ApplyRulesStartSuccess) until the second (GetApplyRulesStatus) comes back...? If so, why is that useful to do here.

Reply:
@msorens I'm using the switchMap to dispatch `ApplyRulesStartSuccess` and `GetApplyRulesStatus` actions simultaneously.

`GetApplyRulesStatus` is just so that we immediately fetch the latest update status after successfully starting an update. This isn't necessary, we could instead just wait for the ongoing 5 sec interval to eventually update the status. 

`ApplyRulesStartSuccess` currently isn't used by anything, I just included the corresponding `Success` and `Error` actions to be consistent with other async actions.